### PR TITLE
API - get plugin settings

### PR DESCRIPTION
**Pull request by npracht**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | x 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/89 
| Issues addressed (#s or URLs) | 
| BC breaks? | none 
| Deprecations? | none 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR create a new endpoint that allows to get the plugin mapping settings via API. It is very useful for a third application to know how plugins are set. 

The API answer looks like as follow:
```json
{ 
   "objects": [
        "lead",
        "company"
    ],
    "update_mautic": {
        "alias_from_integration": "1",
        "alias_from_integration": "1",
        "FirstName": "1"
    },
    "leadFields": {
        "alias_from_integration": "alias_mapped_from_mautic",
        "alias_from_integration": "alias_mapped_from_mautic",
        "FirstName": "firstname"
    },
    "update_mautic_company": {
        "CompanyName": "1"
    },
    "companyFields": {
        "CompanyName": "companyname"
    }
}
```

#### Steps to test this PR:
1. Configure any integration like Salesforce for instance
2. Enable and set field mapping (contact and/or companies)
3. Request `/api/plugins/settings/{integration_name}`
4. Get your answer

### CODE REVIEW

File: app/bundles/PluginBundle/Controller/PluginController.php [hunk 1]
```diff
@@ -608,4 +608,12 @@ public function reloadAction()
             ]
         );
     }
+
+    public function viewSettingsAction($name)
+    {
+        $integrationHelper = $this->factory->getHelper('integration');
```
Comment (alanhartless) [inline app/bundles/PluginBundle/Controller/PluginController.php]: MauticFactory is deprecated and shouldn't be used. Use `$this->get('mautic.helper.integration')` instead.

File: app/bundles/PluginBundle/Controller/PluginController.php [hunk 2]
```diff
@@ -608,4 +608,12 @@ public function reloadAction()
...
+        $integrationObject = $integrationHelper->getIntegrationObject($name);
+        $featureSettings = $integrationObject->getIntegrationSettings()->getFeatureSettings();
+        return new JsonResponse($featureSettings);
```
Comment (alanhartless) [inline app/bundles/PluginBundle/Controller/PluginController.php]: Let's put an empty line before this as per Symfony coding standards.

**Comment by alanhartless** [comment]
This works as described. Let's fix the couple code issues before we can merge.

**Comment by javjim-mautic** [comment]
Works properly

**Comment by npracht** [comment]
@alanhartless is that good now ?
Thanks !

**Comment by escopecz** [comment]
There is a code style issue: https://travis-ci.org/mautic/mautic/jobs/359282707#L702

**Comment by npracht** [comment]
@escopecz i think we're good now ? :)

**Comment by escopecz** [comment]
If I try to call `/api/plugins/settings/twitter` I get this:

```
{
  "errors": [
    {
      "message": "Call to a member function getIntegrationSettings() on boolean",
      "code": 500,
      "type": null
    }
  ],
  "error": {
    "message": "Call to a member function getIntegrationSettings() on boolean (`error` is deprecated as of 2.6.0 and will be removed in 3.0. Use the `errors` array instead.)",
    "code": 500
  },
  "trace": [
    {
      "namespace": "",
      "short_class": "",
      "class": "",
      "type": "",
      "function": "",
      "file": "\/Users\/jan\/dev\/mautic\/app\/bundles\/PluginBundle\/Controller\/PluginController.php",
      "line": 618,
      "args": []
    },
    {
      "namespace": "Mautic\\PluginBundle\\Controller",
      "short_class": "PluginController",
      "class": "Mautic\\PluginBundle\\Controller\\PluginController",
      "type": "-\u003E",
      "function": "viewSettingsAction",
      "file": "\/Users\/jan\/dev\/mautic\/vendor\/symfony\/http-kernel\/HttpKernel.php",
      "line": 135,
      "args": [
        [
          "string",
          "twitter"
        ]
      ]
    },
...
```
Also the new endpoint must be added to https://github.com/mautic/api-library and covered by functional test(s) there.

Then it must be documented here: https://github.com/mautic/developer-documentation

**Comment by npracht** [comment]
@escopecz the documentation is already pulled --> https://github.com/mautic/developer-documentation/pull/89

**Comment by escopecz** [comment]
Amazing! Missed that.

**Comment by mleffler** [comment]
Hi - thanks for the documentation.  If the error is fixed, and the API library is updated, we can get this into 2.13.  We need to be done with all merges by Wednesday, April 4, so finishing by Tuesday would be needed, thank you.  If that won't be possible, let us know and we will push it to the next release.

**Comment by npracht** [comment]
Hello Melissa, i won't be able to work on it before several days.
Thanks for your message.

N.

**Comment by npracht** [comment]
@escopecz `twitter` should be written `Twitter` :)
We're currently working on API library including UT :)

btw - ETA for 2.14.0?

**Comment by npracht** [comment]
@escopecz @alanhartless all modifications done and API+test done :)

**Comment by kuzmany** [comment]
Tested and works properly.
Added validation and notFound response If integration not exist or is disabled https://github.com/mautic/mautic/pull/5306/commits/ca6d56e7b3cc3a6eca2963bf61109595915d4d52

**Comment by escopecz** [comment]
Travis is failing. Please update to current staging. The failing test is fixed there.

**Comment by dbhurley** [comment]
You gotta squash that or rebase. Don't try to merge in 2,036 PR's

**Pull request closed without merging**